vector_search/utils.py

```diff
@@ -214,7 +196,146 @@ def _chunk_documents(encoder, texts, metadatas):
     return recursive_splitter.create_documents(texts=texts, metadatas=metadatas)
 
 
+def _process_resource_embeddings(serialized_resources):
+    docs = []
+    metadata = []
+    ids = []
+    encoder = dense_encoder()
+    vector_name = encoder.model_short_name()
+    for doc in serialized_resources:
+        vector_point_key = doc["readable_id"]
+        metadata.append(doc)
+        ids.append(vector_point_id(vector_point_key))
+        docs.append(
+            f"{doc.get('title')} {doc.get('description')} {doc.get('full_description')}"
+        )
+    if len(docs) > 0:
+        embeddings = encoder.embed_documents(docs)
+        return points_generator(ids, metadata, embeddings, vector_name)
+    return None
+
+
+def generate_metadata_document(serialized_resource):
+    """
+    Generate a plaint-text info document to embed in the contentfile collection
```

Comment:
Can this be replaced with a serializer that takes a learning resource? Something like ContentFileLearningResourceMetadataSerializer

Reply:
👍 That might be a cleaner way to do this. will have a look